packages/fdc3-agent-proxy/src/intents/DefaultIntentSupport.ts

```diff
@@ -163,26 +169,38 @@ export class DefaultIntentSupport implements IntentSupport {
       meta: this.messaging.createMeta(),
     };
 
-    const resultPromise = this.createResultPromise(messageOut);
-    const response: RaiseIntentForContextResponse = await this.messaging.exchange(
-      messageOut,
+    const resultPromise = this.createResultPromise(request);
+    const response = await this.messaging.exchange<RaiseIntentForContextResponse>(
+      request,
       'raiseIntentForContextResponse',
       ResolveError.IntentDeliveryFailed
     );
+
+    throwIfUndefined(
+      response.payload.error ?? response.payload.appIntents ?? response.payload.intentResolution,
```

Comment:
won't get here for a payload error as that will get consumed and thrown by exchange

Reply:
True, someone breaking exchange is hopefully unlikely.

```suggestion
      response.payload.appIntents ?? response.payload.intentResolution,
```